app/code/core/Mage/Core/Model/App.php

```diff
@@ -364,6 +364,23 @@ public function run($params)
 
             $this->getFrontController()->dispatch();
         }
+
+        // Finish the request explicitly, no output allowed beyond this point
+        if (php_sapi_name() == 'fpm-fcgi' && function_exists('fastcgi_finish_request')) {
+            fastcgi_finish_request();
+        } else {
+            flush();
+        }
+        if (isset($_SESSION)) {
```

Comment:
shouldn't we use https://www.php.net/manual/en/function.session-status.php here as @Flyingmana suggested?

Reply:
https://github.com/OpenMage/magento-lts/pull/3145